### PR TITLE
Add check for existence of repository/schema for each request

### PR DESCRIPTION
**Pull request by joswarmer**
Add check for each request whether the repository for the request exists.
Also activated `noFallthroughCasesInSwitch` and `noImplicitAny` TS options, leading to `@ts-expect-error` for database operations as they always return untypes information.

### CODE REVIEW

File: packages/common/src/apiutil/LionwebResponse.ts [hunk 1]
```diff
@@ -30,7 +30,9 @@ export type ResponseMessage = {
  * @param object
  */
 export function isResponseMessage(object: unknown): object is ResponseMessage {
-    return object["kind"] !== undefined && typeof object["kind"] === "string"
+    // @ts-expect-error TS7053
+    return object["kind"] !== undefined && typeof object["kind"] === "string" &&
```
Comment (ftomassetti) [inline packages/common/src/apiutil/LionwebResponse.ts:34]: would casting to `unknown` help avoiding the TS error?

File: packages/common/src/database/DbConnection.ts [hunk 2]
```diff
@@ -76,6 +78,7 @@ export class DbConnection {
      */
     async one(repositoryData: RepositoryData, query: string) {
         query = addRepositorySchema(query, repositoryData)
+        console.log("query: ", query)
```
Comment (ftomassetti) [inline packages/common/src/database/DbConnection.ts:81]: Should we log this one differently?

File: packages/dbadmin/src/controllers/DBAdminApi.ts [hunk 3]
```diff
@@ -55,7 +55,12 @@ export class DBAdminApiImpl implements DBAdminApi {
     createDatabase = async(request: e.Request, response: e.Response) => {
         logger.requestLog(` * createDatabase request received, with body of ${request.headers["content-length"]} bytes`)
         const result = await this.ctx.dbAdminApiWorker.createDatabase(CREATE_DATABASE_SQL)
-        lionwebResponse(response, result.status, {
+        // lionwebResponse(response, result.status, {
```
Comment (ftomassetti) [inline packages/dbadmin/src/controllers/DBAdminApi.ts:58]: Can we remove this code or is this kept in a comment for a reason?

File: packages/dbadmin/src/controllers/DBAdminApi.ts [hunk 4]
```diff
@@ -55,7 +55,12 @@ export class DBAdminApiImpl implements DBAdminApi {
     createDatabase = async(request: e.Request, response: e.Response) => {
         logger.requestLog(` * createDatabase request received, with body of ${request.headers["content-length"]} bytes`)
         const result = await this.ctx.dbAdminApiWorker.createDatabase(CREATE_DATABASE_SQL)
```
Comment (ftomassetti) [inline packages/dbadmin/src/controllers/DBAdminApi.ts:57]: I would consider renaming this one to `createDbResult`

File: packages/dbadmin/src/controllers/DBAdminApi.ts [hunk 5]
```diff
@@ -55,7 +55,12 @@ export class DBAdminApiImpl implements DBAdminApi {
...
-        lionwebResponse(response, result.status, {
+        // lionwebResponse(response, result.status, {
+        //     success: result.status === HttpSuccessCodes.Ok,
+        //     messages: [ {kind: "Info", message: result.queryResult} ]
+        // })
+        const result2 = await this.ctx.dbConnection.query({clientId: "Repository", repository: "public"}, removeNewlinesBetween$$(CREATE_GLOBALS_SQL))
```
Comment (ftomassetti) [inline packages/dbadmin/src/controllers/DBAdminApi.ts:62]: I would consider renaming this one to `createGlobalFunctionsResult`

**Review by ftomassetti — APPROVED** [review]
It looks good to me.

I am not 100% sure if it is best to remove the extra check and spare the `// @ts-expect-error TS7053` lines or keep it. It is difficult for me to assess which option is best.

**Comment by joswarmer** [comment]
> It looks good to me.
> 
> I am not 100% sure if it is best to remove the extra check and spare the `// @ts-expect-error TS7053` lines or keep it. It is difficult for me to assess which option is best.

Because we get a lot of untyped data from the Postgres queries, I see no other way that using @ts-expect-error. For the rest of the code, the stricter checking does help.

File: packages/common/src/apiutil/LionwebResponse.ts (hunk 1 above)
Comment (joswarmer) [reply packages/common/src/apiutil/LionwebResponse.ts:34]: Not really, as you cannot access properties on an unknown object.  And object itself is already unknown.
But you gave me an idea, I can cast the object to ResponseMessage and then access its properties,

File: packages/common/src/database/DbConnection.ts (hunk 2 above)
Comment (joswarmer) [reply packages/common/src/database/DbConnection.ts:81]: yes we should.  I am currently revising the logging and make it more flexible. No console logs anymore and adding log levels, and multiple outputs (I am using console and file output, as the file make it easier to find and collapse messages.
Will leave it for now, next PR is will change.

File: packages/dbadmin/src/controllers/DBAdminApi.ts (hunk 3 above)
Comment (joswarmer) [reply packages/dbadmin/src/controllers/DBAdminApi.ts:58]: Can be removed.

File: packages/dbadmin/src/controllers/DBAdminApi.ts (hunk 4 above)
Comment (joswarmer) [reply packages/dbadmin/src/controllers/DBAdminApi.ts:57]: Removed the variable, as the call either is ok, or throws an exception. Result is never useful.

File: packages/dbadmin/src/controllers/DBAdminApi.ts (hunk 5 above)
Comment (joswarmer) [reply packages/dbadmin/src/controllers/DBAdminApi.ts:62]: Indeed.

Comment (joswarmer) [reply packages/dbadmin/src/controllers/DBAdminApi.ts:62]: Removed this variable as well, as there is no return value, only an exception if something goes wrong.

**Pull request merged**